md-proto/WebAssembly.md

```diff
@@ -0,0 +1,2007 @@
+limit value is zero.
+
+> The final `end` instruction may be implicit in some representations.
+
+#### Function Body Validation
+
+The instruction at the current position is remembered, and the current position
+is incremented to point to the position following it. Then the remembered
+instruction is validated as follows.
+
+If the instruction has a **Validation** clause, that clause describes its
+validation. Otherwise, [generic validation](#generic-instruction-validation) is
+performed.
+
+If the current position is now past the end of the sequence,
+[function return validation](#function-return-validation) is initiated and
+validation of the function is thereafter complete.
+
+Otherwise, [validation](#function-body-validation) is resumed.
+
+TODO: `unreachable`, `br`, `return`, and `br_table` return type validation.
+
+#### Generic Instruction Validation
+
+For each operand [type] in the instruction's signature in reverse order, a
+type is popped from the type stack and checked to match it. Each return type of
+the instruction's signature is then pushed onto the type stack.
+
+Then, if the instruction's signature has no return types, and the length of the
+type stack is greater than the control-flow stack top's limit value, validation
+fails.
+
+#### Function Return Validation
+
+If the function signature has any return types, a type is popped from the type
+stack and checked to match the first return type. The type stack and the
+control-flow stack are then both checked to be empty. If no failures were
+detected anywhere in the function, function validation is successful.
+
+
+Execution
+--------------------------------------------------------------------------------
+
+Before any code in a module can be executed, the whole module must first be
+[validated](#validation).
+
+The contents of the [Data Section](#data-section) are loaded into linear memory.
+Each [string] is loaded into linear memory at its associated offset.
+
+Then, the [recursion limit](#recursion-limits) values are chosen,
+[nondeterministically].
+
+TODO: Describe module versus instance, and describe instantiation.
+
+TODO: Implementations can, nondeterministically, trap at any time, due to
+insufficient resources.
+
+### Recursion Limits
+
+The following two values are chosen before any functions are executed:
+
+ - A *recursion increment*.
+ - A *recursion limit*.
+
+> These values are used by [call instructions][L].
+
+### Module Execution
+
+If the module contains a [Start Section](#start-section), the referenced
+function is [executed](#function-execution).
+
+### Function Execution
+
+Function execution can be prompted by a [call-family instruction][L], by
+[module execution](#module-execution), or by the embedding environment.
+
+The input to execution of a function consists of:
+ - The function to be executed.
+ - The incoming argument values, one for each argument [type] of the function.
+ - A *recursion depth* value.
+
```

Comment:
The other mentions of the phrase "recursion depth" are meant to explain what it's used for.